### DISCUSSION TITLE
Any plans for cell spanning in DataTable

### DISCUSSION THREAD

**Discussion by leemthai** [Textualize/textual#5236] [Ideas]
Would make for a great feature I think. I certainly need it haha

**Comment by leemthai** [comment]
I guess if I have to code it myself, I would do something like:

- Set cell_padding to zero to allow for possibility of padding
- For rows I want to display columns without spans then only use column_width-1 characters, leave the right-hand character as 'virtual cell padding'
-  For rows I want to span text... hmm... really not sure how easy or difficult it is to span formatted (linked?) text across cells?

Is that a reasonable starting idea?

**Comment by TomJGooding** [comment]
Depending on what you're trying to achieve, perhaps it would be easier to build a widget akin to a table using the [grid layout](https://textual.textualize.io/guide/layout/#grid), rather than trying to fight the `DataTable` defaults for rows and columns?

**Comment by leemthai** [comment]
That's a very interesting idea. I am fighting the layout restrictions of `DataTable`, somewhat.   I guess the big concern is how efficient, in terms of memory but mainly in terms of update speed, using 100-200 widgets in a grid would be, compared with a single  `DataTable` widget. Are there many apps around that use lots of small widgets?

**Reply by TomJGooding** [reply]
You're right that it wouldn't be as performant, but it might not be noticeable with only 100-200 widgets.

I quickly created a contrived example of a number grid, where incrementing all the numbers updates pretty much instantly:

```python
from textual.app import App, ComposeResult
from textual.widget import Widget
from textual.widgets import Footer, Static


class Cell(Static):
    DEFAULT_CSS = """
    Cell {
        width: auto;
        padding: 0 1;
    }
    """


class NumberGrid(Widget, can_focus=True):
    BINDINGS = [("+", "increment", "Increment")]

    DEFAULT_CSS = """
    NumberGrid {
        height: auto;
        max-height: 100%;
        background: $surface;
        layout: grid;
        grid-size: 10;
        grid-columns: auto;
    }
    """

    starting_num = 1
    total_nums = 200

    def compose(self) -> ComposeResult:
        for num in range(
            self.starting_num,
            (self.starting_num + self.total_nums),
        ):
            yield Cell(str(num))

    def action_increment(self) -> None:
        self.starting_num += 1
        cells = self.query(Cell)
        for idx, num in enumerate(
            range(
                self.starting_num,
                (self.starting_num + self.total_nums),
            )
        ):
            cells[idx].update(str(num))


class ExampleApp(App):
    def compose(self) -> ComposeResult:
        yield NumberGrid()
        yield Footer()


if __name__ == "__main__":
    app = ExampleApp()
    app.run()
```

**Comment by leemthai** [comment]
Yeah, I suspect the performance hit will be fairly minor, though even the demo (upped to 15x35 cells) does seem to be spinning my fan up , which is never a great sign. I will update with some timings and further thoughts soon.
